File: mobile/library/kotlin/src/io/envoyproxy/envoymobile/StatsClientImpl.kt
package io.envoyproxy.envoymobile

import io.envoyproxy.envoymobile.engine.EnvoyEngine

/**
 * Envoy implementation of `StatsClient`.
 */
internal class StatsClientImpl constructor(
  internal val engine: EnvoyEngine
) : StatsClient {

  override fun counter(vararg elements: Element): Counter {
    return Counter(engine, elements.asList())
  }
}

File: mobile/library/swift/src/StatsClientImpl.swift
@_implementationOnly import EnvoyEngine
import Foundation

/// Envoy implementation of StatsClient.
final class StatsClientImpl: NSObject {
  private let engine: EnvoyEngine

  init(engine: EnvoyEngine) {
    self.engine = engine
    super.init()
  }
}

extension StatsClientImpl: StatsClient {
  func counter(elements: [Element]) -> Counter {
    return CounterImpl(elements: elements, engine: self.engine)
  }
}
